src/Microsoft.IdentityModel.Clients.ActiveDirectory/Features/WinCommon/AuthenticationContext.cs

```diff
@@ -36,314 +36,273 @@ namespace Microsoft.IdentityModel.Clients.ActiveDirectory
     /// <summary>
     /// Extension class to support confidential client flows.
     /// </summary>
-    public static class AuthenticationContextConfidentialClientExtensions
+    public sealed partial class AuthenticationContext
     {
 #if !WINDOWS_APP
 
         /// <summary>
         /// Gets URL of the authorize endpoint including the query parameters.
         /// </summary>
-        /// <param name="ctx">Authentication context instance</param>
         /// <param name="resource">Identifier of the target resource that is the recipient of the requested token.</param>
         /// <param name="clientId">Identifier of the client requesting the token.</param>
         /// <param name="redirectUri">Address to return to upon receiving a response from the authority.</param>
         /// <param name="userId">Identifier of the user token is requested for. This parameter can be <see cref="UserIdentifier"/>.Any.</param>
         /// <param name="extraQueryParameters">This parameter will be appended as is to the query string in the HTTP authentication request to the authority. The parameter can be null.</param>
         /// <returns>URL of the authorize endpoint including the query parameters.</returns>
-        public static async Task<Uri> GetAuthorizationRequestUrlAsync(this AuthenticationContext ctx, string resource,
+        public async Task<Uri> GetAuthorizationRequestUrlAsync(string resource,
             string clientId, Uri redirectUri,
             UserIdentifier userId, string extraQueryParameters)
         {
-            return await GetAuthorizationRequestUrlAsync(ctx, resource, clientId, redirectUri, userId,
+            return await GetAuthorizationRequestUrlAsync(resource, clientId, redirectUri, userId,
                 extraQueryParameters,
-                null);
+                null).ConfigureAwait(false);
```

Comment:
@kpanwar is the addition of .ConfigureAwait(false) directly related to the hotfix?

Reply:
@jmprieur - not but it should have been there. I just caught it as I was cleaning up the code.

Reply:
Thanks for confirming @kpanwar

Reply:
Tip: you can guard against accidental misses like this by installing the `AsyncUsageAnalyzers` and then setting the `ConfigureAwait` analyzer to Error (or warning, or whatever you want). It's a life saver for library development.